modules/RedisTimeSeriesEdge/Program.cs

```diff
@@ -80,6 +91,19 @@ static async Task Init()
             await moduleClient.SetMethodDefaultHandlerAsync(GetTimeSeriesInfo, moduleClient);
         }
 
+        static async Task ReportAssemblyVersionAsync(ModuleClient moduleClient)
+        {
+            var assemblyVersion = Assembly.GetEntryAssembly().GetName().Version.ToString();            
+            Log.LogInformation($"Assembly Version: {assemblyVersion}");
+
+            var reportedProperties = new TwinCollection
+            {
+                ["AssemblyVersion"] = assemblyVersion
+            };
+
+            await moduleClient.UpdateReportedPropertiesAsync(reportedProperties);

```

Comment:
in case the module already has some properties, wouldn't we override them with just `reportedProperties`? Shouldn't we first fetch any existing properties, and append them with new properties?
I may be wrong, maybe `moduleClient` already handles it.

Reply:
Initially I thought so too but it seems this is the way it works according to our [C# examples](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-csharp-csharp-module-twin-getstarted#update-the-module-twin-using-net-device-sdk). I tried loading the module twin first and adding my property on top, but got the following exception
```cs
System.InvalidOperationException: Property name $version contains invalid character '$'
at Microsoft.Azure.Devices.Edge.Hub.Core.Twin.ReportedPropertiesValidator.ValidatePropertyNameAndLength(String name)
```